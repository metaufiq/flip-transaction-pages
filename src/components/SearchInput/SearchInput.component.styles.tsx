import { StyleSheet } from "react-native";

const styles = StyleSheet.create({
  container: {
    backgroundColor: 'white',
    borderRadius: 1,
    paddingHorizontal: 10,
    paddingVertical: 20
  }
})

export default styles;