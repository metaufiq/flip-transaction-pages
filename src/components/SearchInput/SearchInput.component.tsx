import React from "react";
import { Text, TextInput, View } from "react-native";

import styles from "./SearchInput.component.styles";
import { Props } from "./SearchInput.component.types";

const SearchInput = ({placeholder, onChange}: Props)=>{
  return(
    <View style={styles.container}>
      <TextInput 
        placeholder={placeholder}
        onChangeText={onChange}
      />
    </View>
  )
}

export default SearchInput;