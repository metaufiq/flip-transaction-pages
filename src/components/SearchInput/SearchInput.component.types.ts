export interface Props{
  placeholder?: string,
  onChange: (value:string)=>void
}