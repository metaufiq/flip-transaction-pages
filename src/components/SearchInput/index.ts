import SearchInput from "./SearchInput.component";

export default SearchInput