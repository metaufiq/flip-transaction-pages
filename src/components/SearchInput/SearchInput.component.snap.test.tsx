import { assertSnapshots } from "../../../test";
import SearchInput from ".";
import { Props } from "./SearchInput.component.types";

describe('SearchInputSnapshot', () => {

  const mockProps:Props = {
    onChange: ()=>{}
  };

  const configs = [
    {
      props: mockProps,
      desc: 'should render search input'
    },
    {
      props: {...mockProps, placeholder: 'placeholder'},
      desc: 'should render search input with placeholder'
    },
  ];

  assertSnapshots(SearchInput, configs);
});
