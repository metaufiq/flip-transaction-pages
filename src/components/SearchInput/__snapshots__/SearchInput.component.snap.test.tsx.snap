// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchInputSnapshot should render search input 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "borderRadius": 1,
      "paddingHorizontal": 10,
      "paddingVertical": 20,
    }
  }
>
  <TextInput
    onChangeText={[Function]}
  />
</View>
`;

exports[`SearchInputSnapshot should render search input with placeholder 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "borderRadius": 1,
      "paddingHorizontal": 10,
      "paddingVertical": 20,
    }
  }
>
  <TextInput
    onChangeText={[Function]}
    placeholder="placeholder"
  />
</View>
`;
